ql/src/java/org/apache/hadoop/hive/ql/security/authorization/plugin/metastore/HiveMetaStoreAuthorizer.java

```diff
@@ -198,6 +446,29 @@ HiveMetaStoreAuthzInfo buildAuthzContext(PreEventContext preEventContext) throws
     return ret;
   }
 
+  HiveAuthorizer createHiveMetaStoreAuthorizer() throws Exception {
```

Comment:
How costly is this method? Is the Authorizer user specific, or can we have a single instance per HMS thread, or even better on HMS instance (if this stuff is thread-safe?)

Reply:
Authorizer has some user specific config. Because the class was there before, so I will explore changing it to singleton or something in a separate jira for perf reason. I will work on it next. https://issues.apache.org/jira/browse/HIVE-23943